server/src/main/java/org/elasticsearch/indices/NodeIndicesStats.java

```diff
@@ -204,6 +214,11 @@ public DenseVectorStats getDenseVectorStats() {
         return stats.getDenseVectorStats();
     }
 
+    // for testing
+    Map<Index, List<IndexShardStats>> getStatsByShard() {
```

Comment:
Could we test this via `getShardStats(Index index)` instead please?

Reply:
Sure, we test this via `getShardStats(Index index)` in 3ddf821660929f8525424a8ab66abf343a5a5f33